optimum/exporters/ipex/model_patcher.py

```diff
@@ -62,25 +60,11 @@ def patch_op(m, target_m, new_op_name, new_op):
 
 
 def _patch_llama_model(model):
-    if is_ipex_version("<", "2.5.0"):
-        raise ImportError("Only ipex version > 2.3.0 supports RotaryEmbedding and IndirectAccessKVCache")
-
-    from intel_extension_for_pytorch.llm.modules import IndirectAccessKVCache, RotaryEmbedding
-
-    ipex_rope = RotaryEmbedding(
-        model.config.max_position_embeddings,
-        model.config.hidden_size // model.config.num_attention_heads,
-        model.config.rope_theta,
-        model.config.architectures[0],
-    )
-    ipex_scale_dot_product = IndirectAccessKVCache(text_max_length=model.config.max_position_embeddings)
-    patch_op(model, LlamaAttention, "ipex_rope", ipex_rope)
-    patch_op(model, LlamaAttention, "ipex_scale_dot_product", ipex_scale_dot_product)
+    if is_ipex_version("<", "2.3.0"):
+        raise ImportError("Only ipex version >= 2.3.0 supports llama model patching")
 
     convert_functions(model, LlamaModel, "forward", _llama_model_forward)
-    convert_functions(model, LlamaAttention, "forward", _llama_attn_forward)
     convert_functions(model, LlamaRMSNorm, "forward", _llama_layer_norm_forward)
-
     convert_class(model, LlamaDecoderLayer, _IPEXLlamaDecoderLayerRef, model.config)
```

Comment:
question: since you already convert the whole model's forward, I suppose you can customize rms/decoder in the model's forward. Why we still need convert 2 modules, educate me the optimum patching design on this.

Reply:
rms exists in different submodules in the model. It will be less codes by `convert_functions`.

Reply:
Answer: their is a [rms](https://github.com/huggingface/transformers/blob/v4.38.2/src/transformers/models/llama/modeling_llama.py#L1037) layer which is post all decoding layer, which cannot be covered by the DecoderLayer conversion. So need a specific conversion.